src/sage/knots/link.py

```diff
@@ -3121,26 +3115,44 @@ def is_colorable(self, n):
             sage: K8.is_colorable(3)
             False
 
+        But it is colorable with respect to the value of its determinant::
+
+            sage: K8.determinant()
+            5
+            sage: K8.is_colorable()
+            True
+
+        An examples with non prime determinant::
+
+            sage: K = Knots().from_table(6, 1)
+            sage: K.determinant()
+            9
+            sage: K.is_colorable()
+            True
+
         REFERENCES:
 
         - :wikipedia:`Fox_n-coloring`
 
         - Chapter 3 of [Liv1993]_
 
-        .. SEEALSO:: :meth:`colorings`
+        .. SEEALSO:: :meth:`colorings` and :meth:`coloring_maps`
         """
-        return self._coloring_matrix(n).nullity() > 1
+        try:
```

Comment:
Is there a reason to try to use `nullity` instead of going directly for the way that always works?

Reply:
My intention is to maintain performance on prime numbers. It would decrease by a factor of 28:

```
sage: KnotInfo.K12a_1007.inject()
Defining K12a_1007
sage: K12a_1007.determinant()
143
sage: K = K12a_1007.link()
sage: M = K._coloring_matrix(n=13)
sage: %timeit M.nullity()
1.47 µs ± 169 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
sage: %timeit M.right_kernel_matrix().dimensions()[0]
41.5 µs ± 1.09 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

For just one call this is negligible. But if invoked inside a large loop this could count.

Reply:
In that case, maybe checking for the ring to be a field would be better than trying and catching the exception.

Reply:
> In that case, maybe checking for the ring to be a field would be better than trying and catching the exception.

I agree! I just made the change.